src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -225,6 +320,46 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
         DenseMatrix states = solveTransposed(rhs, j);
 
         // calculate sensitivity values
-        return calculateSensitivityValues(lfNetwork, equationSystem, factorsByVarConfig, states);
+        List<SensitivityValue> sensitivityValues = calculateSensitivityValues(lfNetwork, equationSystem, factorsByVarConfig, states);
+
+        if (!lfParameters.isDistributedSlack()) {
+            return sensitivityValues;
```

Comment:
Having a return in the middle of the function makes it harder to understand, I need to find a better way to distinguish the distributed/ not distributed case